Loader/SymfonyFixturesLoader.php

```diff
@@ -65,4 +83,70 @@ protected function createFixture($class)
 
         return $this->loadedFixtures[$class];
     }
+
+    /**
+     * Returns the array of data fixtures to execute.
+     *
+     * @param string[] $groups
+     *
+     * @return FixtureInterface[]
+     */
+    public function getFixtures(array $groups = [])
+    {
+        $fixtures = parent::getFixtures();
+
+        if (empty($groups)) {
+            return $fixtures;
+        }
+
+        $filteredFixtures = [];
+        foreach ($fixtures as $fixture) {
+            foreach ($groups as $group) {
+                $fixtureClass = get_class($fixture);
+                if (isset($this->groupsFixtureMapping[$group][$fixtureClass])) {
+                    $filteredFixtures[$fixtureClass] = $fixture;
+                    continue 2;
+                }
+            }
+        }
+
+        $loadedFixtureClasses = array_keys($filteredFixtures);
+        foreach ($filteredFixtures as $fixture) {
+            $this->validateDependencies($loadedFixtureClasses, $fixture);
+        }
+
+        return array_values($filteredFixtures);
+    }
+
+    /**
+     * Generates an array of the groups and their fixtures
+     *
+     * @param string[] $groups
+     */
+    private function addGroupsFixtureMapping(string $className, array $groups): void
+    {
+        foreach ($groups as $group) {
+            $this->groupsFixtureMapping[$group][$className] = true;
+        }
+    }
+
+    /**
+     * @param string[] $loadedFixtureClasses
+     *
+     * @throws RuntimeException
+     */
+    private function validateDependencies(array $loadedFixtureClasses, FixtureInterface $fixture): void
+    {
+        if (! $fixture instanceof DependentFixtureInterface) {
+            return;
+        }
+
+        $dependenciesClasses = $fixture->getDependencies();
+
+        foreach ($dependenciesClasses as $class) {
+            if (!in_array($class, $loadedFixtureClasses)) {
```

Comment:
Better to use `isset` or `array_key_exists`. You can do that if you pass the loaded fixture classes in the array keys instead of in the array values.